srcs/app/config/analyser/Token.cpp

```diff
@@ -0,0 +1,94 @@
+#include <parser/Token.hpp>
+
+std::string Token::Type::GetTokenTypeStr(enum e_id type) {
+	std::string retval("unknown");
+	if (type > T_UNKNOWN && type < T_INVALID)
+		return str_map[type];
+	return retval;
+}
+
+const char *Token::Type::str_map[10] = {
+	"t scope open",
+	"t scope close",
+	"t symbol",
+	"t str immediate t0",
+	"t str immediate t1",
+	"t int immediate",
+	"t uint immediate",
+	"t dbl immediate",
+	"t end"
+};
+
+std::string Token::State::GetParsingStateTypeStr(enum e_id type) {
+	std::string retval("");
+	if (type > K_NONE && type < K_LAST_INVALID_STATE)
+		return keyword_to_str[type];
+	return retval;
+}
+
+t_parsing_state Token::State::GetParsingStateTypeEnum
+(const std::string &data, t_token_type ttype) {
+	if (ttype == Token::Type::T_END || ttype == Token::Type::T_SCOPE_OPEN
+		|| ttype == Token::Type::T_SCOPE_CLOSE)
+		return K_EXP_KW;
+	for (uint8_t i = K_EXIT; i < K_LAST_INVALID_STATE; ++i) {
+		if (keyword_to_str[i + 1] == data)
+			return static_cast<t_parsing_state>(i);
+	}
+	return K_NONE;
+}
+
+const char *Token::State::keyword_to_str[20] = {
+	"",
+	"EXIT",
+	"INIT",
+	"server",
+	"listen",
+	"server_name",
+	"root",
+	"client_max_body_size",
+	"error_page",
+	"return",
+	"autoindex",
+	"index",
+	"upload_store",
+	"return_status",
+	"return_url",
+	"cgi_assign",
+	"location",
+	"limit_except",
+	"EXPECT SEMICOLON",
+	"EXPECT KEYWORD"
+};
+
+Token::Token(const std::string &data, t_token_type type, size_t line)
+	: data_(data),
+	  type_(type),
+	  line_(line),
+	  event_(ParsingEvents::GetEvent(*this)),
+	  state_(State::GetParsingStateTypeEnum(data_, type_)) {
+}
```

Comment:
I would say the parser class is already way overcrowded.
I may decide later on to do as you say, but I'd prefer to keep it this way for now, to have the full picture as it is in Parser. Do you mind? Every small change now equals to a lot of changes, and I'm not sure your solution really is the best. I used to have two different classes for events and for states, but I decided to flatten it all into Tokens, since all of the data can be derived from it.

Reply:
Ok, sounds good :+1: